### PR TITLE
Fix #481 - add reasons to stattype guesses and make schema guessing r…

### LINKED ISSUE #481: Improve handling of reasons in GUESS SCHEMA
Currently, reasons for stattype guesses are encoded as comments enclosed in triple quotes.
This format assumes the user will be using a triple quote string within a bdb.execute() command to create a population with the schema. 
With quotes, the schema text cannot be directly copy/pasted into an MML block with iVenture. There seems to be no way to include the reasons such that an MML cell in a Jupyter notebook could ignore them on execution. 

Two potential plans for improving this: 
1. Modify MML such that there are two options: `GUESS SCHEMA FOR <table>` and `GUESS SCHEMA FOR <table> WITH REASONS`.
2. Make `GUESS SCHEMA FOR <table>` create a new table in the bdb with three columns: column, guessed_stattype, and reason. Then, the user could edit stattypes by modifying the table. We would need to add an additional MML command such as `PRINT SCHEMA STRING` to convert the table to a string that could be copied and pasted into a `CREATE POPULATION` command.

### PR DESCRIPTION
**Pull request by curlette**
…eturn a table.

Add reasons to stattype guesses.

Edit population guessing code to handle new stattype guessing.

Fix failing tests.

Edit nominal reason.

### CODE REVIEW

File: src/guess.py [hunk 1]
```diff
@@ -202,30 +204,34 @@ def bayesdb_guess_stattypes(column_names, rows, null_values=None,
...
                                              numcat_ratio=numcat_ratio,
                                              have_key=(key is not None))
             if stattype == 'key':
                 key = column_name
-        stattypes.append(stattype)
+        stattypes.append([stattype, reason])
     return stattypes
 
-def guess_column_stattype(column, **kwargs):
+def guess_column_stattype(column, reason='', **kwargs):
```
Comment (fsaad) [inline src/guess.py]: Why is `reason=''` and argument to this function? Where is it being used by a client?

File: src/guess.py [hunk 2]
```diff
@@ -202,30 +204,34 @@ def bayesdb_guess_stattypes(column_names, rows, null_values=None,
...
         del counts[None]
     if len(counts) < 2:
-        return 'ignore'
+        return ['ignore', '%s There is only one unique value.' %(reason,)]
     (most_numerous_key, most_numerous_count) = sorted(
         counts.items(), key=lambda item: item[1], reverse=True)[0]
     if most_numerous_count / float(len(column)) > kwargs['nullify_ratio']:
         column = [ None if v == most_numerous_key else v for v in column ]
-        return guess_column_stattype(column, **kwargs)
+        return guess_column_stattype(column, '%s More than %d percent of the '
```
Comment (fsaad) [inline src/guess.py]: I see, `reason` is used in the recursive call, although I am confused about this behavior.

File: src/guess.py (hunk 1 above)
Comment (curlette) [inline src/guess.py]: `guess_column_stattype` is recursive, so I wanted to be able to append to the reason on a recursive call.

Comment (fsaad) [inline src/guess.py]: Right, as noted in some other review comment which appears to have disappeared into github's `/dev/null`... I later noticed the recursive call and was confused about why it was necessary.

**Pull request merged**